lib/plugins/aws/provider.js

```diff
@@ -1806,6 +1846,17 @@ class AwsProvider {
       );
     }
   }
+
+  reportDeprecatedUsage(deprecationCode, pairs) {
+    for (const [deprecated, modern] of pairs) {
+      if (_.has(this.serverless.service.provider, deprecated)) {
+        this.serverless._logDeprecation(
```

Comment:
Deprecation logger works in a way that it reports given deprecation (code) only once.

So with the way it's constructed now, all iteration after first log, is ineffective.

Best way would be to first gather info of all deprecated properties, and then list all of them with one log.

Reply:
Not sure it's fair a comparison, as there's still going to be a loop, and still at least one call to `this.serverless._logDeprecation`. Up to you really 🤷‍♂️

Reply:
Point, is to inform about all deprecated properties with single `logDeprecation` call (yes, we still need to for loop, that doesn't contradict that)